fastcrypto/src/traits.rs

```diff
@@ -110,6 +110,24 @@ pub trait VerifyingKey:
             .try_for_each(|(pk, sig)| pk.verify(msg, sig))
             .map_err(|_| eyre!("Signature verification failed"))
     }
+
+    // Expected to be overridden by implementations
+    fn verify_batch_empty_fail_different_msg<'a, M, I>(msgs: I, pks: &[Self], sigs: &[Self::Sig]) -> Result<(), eyre::Report> where
+        M: Borrow<[u8]> + 'a,
+        I: IntoIterator<Item=M>,
+    {
+        if sigs.is_empty() {
+            return Err(eyre!("Critical Error! This behaviour can signal something dangerous, and that someone may be trying to bypass signature verification through providing empty batches."));
+        }
+        if pks.len() != sigs.len() {
+            return Err(eyre!("Mismatch between number of signatures and public keys provided"));
+        }
+        pks.iter()
+            .zip(sigs)
+            .zip(msgs.into_iter())
```

Comment:
Note that there's no error if the number of messages and signatures/keys do not match. Is this an issue?